src/modules/dependencies.ts

```diff
@@ -1,48 +1,84 @@
-import { execSync } from 'child_process';
+import { execSync } from "child_process";
 
-import SourceInfo from '../utils/lib/SourceInfo';
-import SystemInfo from '../utils/lib/SystemInfo';
-import Syscall from '../utils/lib/Syscall';
-import DependenciesData from '../utils/lib/DependenciesData';
-import HermitOptions from '../utils/lib/HermitOptions'
+import SourceInfo from "../utils/lib/SourceInfo";
+import SystemInfo from "../utils/lib/SystemInfo";
+import Syscall from "../utils/lib/Syscall";
+import DependenciesData from "../utils/lib/DependenciesData";
+import HermitOptions from "../utils/lib/HermitOptions";
 
-import { readDebianPackages, readLanguagePackages } from '../utils/fileSystem';
-import logger from '../utils/logger';
+import { readDebianPackages, readLanguagePackages } from "../utils/fileSystem";
+import logger from "../utils/logger";
+import { runCommandInContainer } from "../utils/containers";
+import * as os from "os";
+
+const processDpkgOutput = (output: string): string | null => {
+  if (output.includes("no path found")) return null;
+
+  return output.split(":")[0];
+};
+
+const getPackageName = async (
+  library: string,
+  image: string,
+  container: boolean
+) => {
+  const shouldRunInContainer = container || os.type() === "Darwin";
```

Comment:
Does this enables the tool to run in a Mac, through the usage of dpkg inside of a container? That’s an interesting approach. What’s the base image in this situation? And how do you you use strace in this scenario?

Reply:
Unfortunately no. I thought that would be enough initially but I forgot `strace` only works on Linux, so it still doesn't work.